Add a section on sourceMappingURL (#151)
WebAssembly/tool-conventions: Debugging.md

may also contain the full wasm file as well, which can be simpler to handle
(and tends to have little downside, as DWARF size tends to be much bigger than
wasm size anyhow).

# Source maps

Adoption of DWARF is a fairly recent addition to WebAssembly, and many toolchains still

> support an earlier [Source Map based debugging proposal](https://github.com/WebAssembly/design/pull/1051).

Reviewer: Since that PR might or might not ever make it into the wasm web spec, maybe we should just copy the relevant bits of it here, so that someone reading this wouldn't have to go read that and figure out what's relevant .That PR is really more of a design doc than a spec (it is a PR against the design repo after all). I think this PR currently has the info from "Linking generated code to source maps", so maybe just copy the info from "Locations"?

Author: I was hesitant to go too deep into details in this doc, just like we don't explain specifics of DWARF here, but only link to external resources, but if it's just the "locations" info, then yeah, it seems reasonably short to include...

Author: Merged the Locations explanation into this description.

This proposal uses a mapping between locations in source files and an offset in the WebAssembly binary
stored in an external file in a format as defined by the [Source Map spec](https://sourcemaps.info/spec.html).

On the WebAssembly side, the URL of this Source Map file is stored in a custom section with the name
`sourceMappingURL`. That section contains:

| Field        | Type        | Description                       |
| ------------ | ----------- | --------------------------------- |
| url_name_len | `varuint32` | Length of `url_name_str` in bytes |

> | url_name_str | `bytes`     | Path to the Source Map file       |

Reviewer 2: ```suggestion
| url_name_str | `bytes`     | Url to the Source Map file       |
```

Author: To go all the way:
```suggestion
| url_name_str | `bytes`     | URL to the Source Map file        |
```

# Source maps

Adoption of DWARF is a fairly recent addition to WebAssembly, and many toolchains still

> support a [Source Map](https://sourcemaps.info/spec.html) based debugging instead.

Reviewer: ```suggestion
support [Source Map](https://sourcemaps.info/spec.html) based debugging as well (or instead).
```
